### PR TITLE
Gpg verification

### LINKED ISSUE #32: Check Downloaded Files
Could we add some checking of downloaded files to the Dockerfiles?

At the moment they do things like:

    RUN set -x \
	&& mkdir -p /usr/src/python \
	&& curl -SL  "https://www.python.org/ftp/python/$PYTHON_VERSION/Python$PYTHON_VERSION.tar.xz" \
		| tar -xJC /usr/src/python --strip-components=1 \

There is no guarantee that this file hasn't been tampered with or corrupted. It would be great if we could test a checksum or signature file prior to unpacking. To see how to download files securely, have a look at the Wordpress and MongoDB Dockerfiles:

https://github.com/docker-library/wordpress/blob/990b1b00b8ca4903e11e53e908b1996fbaab3c1a/Dockerfile
https://github.com/docker-library/mongo/blob/fc66d9cbedac47806c7ae05b1b291c4ee32f6e6a/2.8/Dockerfile

### PR DESCRIPTION
**Pull request by Moghedrin**
Add gpg verification to the Python image.

Fixes #32

### CODE REVIEW

**Comment by tianon** [comment]
I thought we'd agreed to just go with the one person who was the latest release manager for now, especially to avoid the problems with Barry's ancient key?

File: 3.4/Dockerfile [hunk 1]
```diff
@@ -9,10 +9,29 @@ ENV LANG C.UTF-8
...
+	DBBF2EEBF925FAADCF1F3FFFD9866941EA5BBD71 \
+#Short version of the key, due to the fact that HKP keyservers do not support v3 fingerprints
+	7123BC13ED9D77D5
+
 RUN set -x \
 	&& mkdir -p /usr/src/python \
-	&& curl -SL "https://www.python.org/ftp/python/$PYTHON_VERSION/Python-$PYTHON_VERSION.tar.xz" \
-		| tar -xJC /usr/src/python --strip-components=1 \
+	&& curl -SL "https://www.python.org/ftp/python/$PYTHON_VERSION/Python-$PYTHON_VERSION.tar.xz" -o "Python-$PYTHON_VERSION.tar.xz" \
+	&& curl -SL "https://www.python.org/ftp/python/$PYTHON_VERSION/Python-$PYTHON_VERSION.tar.xz.asc" -o "Python-$PYTHON_VERSION.tar.xz.asc" \
```
Comment (tianon) [inline 3.4/Dockerfile]: This could just be `-o python.tar.xz` and `-o python.tar.xz.asc` to simplify these and the following lines.

**Comment by Moghedrin** [comment]
ping @tianon @yosifkit

File: 3.3/Dockerfile [hunk 2]
```diff
@@ -9,10 +9,17 @@ ENV LANG C.UTF-8
 
 ENV PYTHON_VERSION 3.3.6
 
+#Keys from https://www.python.org/downloads/
+RUN gpg --keyserver pool.sks-keyservers.net --recv-keys \
+	26DEA9D4613391EF3E25C9FF0A5B101836580288
```
Comment (tianon) [inline 3.3/Dockerfile]: Georg Brandl (key id: 36580288)

File: 3.4/Dockerfile [hunk 3]
```diff
@@ -9,10 +9,17 @@ ENV LANG C.UTF-8
 
 ENV PYTHON_VERSION 3.4.2
 
+#Keys from https://www.python.org/downloads/
+RUN gpg --keyserver pool.sks-keyservers.net --recv-keys \
+	97FC712E4C024BBEA48A61ED3A5CA953F73C700D
```
Comment (tianon) [inline 3.4/Dockerfile]: Larry Hastings (key id: F73C700D)

File: 2.7/Dockerfile [hunk 4]
```diff
@@ -9,10 +9,17 @@ ENV LANG C.UTF-8
 
 ENV PYTHON_VERSION 2.7.9
 
+#Keys from https://www.python.org/downloads/
+RUN gpg --keyserver pool.sks-keyservers.net --recv-keys \
+	C01E1CAD5EA2C4F0B8E3571504C367C218ADD4FF
```
Comment (tianon) [inline 2.7/Dockerfile]: Benjamin Peterson (key id: 18ADD4FF)

**Comment by tianon** [comment]
(comments in the files plz)

**Comment by Moghedrin** [comment]
Better, @tianon?

**Comment by tianon** [comment]
LGTM

**Comment by yosifkit** [comment]
LGTM

**Pull request merged**